middleware/gzip/filter.go

```diff
@@ -72,7 +84,7 @@ func DefaultMIMEFilter() MIMEFilter {
 // matches any of the registered ones. It returns true if
 // found and false otherwise.
 func (m MIMEFilter) ShouldCompress(r *http.Request) bool {
-	return m.Types.Contains(r.Header.Get("Content-Type"))
+	return m.Types.Contains(r.Header.Get("Accept"))
```

Comment:
I just realised `Accept` flag may consist more info than just `MIME` and can be multiple `MIME`.  I want to suggest we either parse it correctly before checking if it exists in the Set or drop `Accept` and just derive MIME types from file extensions.

I will prefer the former.